File: packages/csharp/src/Errors/NetworkException.cs
using System;

namespace Tesote.Sdk.Errors;

/// <summary>DNS, connection refused, reset — no HTTP response received.</summary>
public sealed class NetworkException : TransportException
{
    /// <summary>Construct from a <see cref="RequestSummary"/> and underlying cause.</summary>
    public NetworkException(string? message, RequestSummary? requestSummary, int attempts, Exception? cause)
        : base(message, "NETWORK_ERROR", 0, null, null, null, null, requestSummary, attempts, cause)
    {
    }
}

File: packages/java/src/main/java/com/tesote/sdk/errors/NetworkException.java
package com.tesote.sdk.errors;

public final class NetworkException extends TransportException {
    public NetworkException(String message, RequestSummary requestSummary,
                            int attempts, Throwable cause) {
        super(message, "NETWORK_ERROR", 0, null, null, null, null,
                requestSummary, attempts, cause);
    }
}
